siliconcompiler/schema/schema_cfg.py

```diff
@@ -3506,6 +3484,177 @@ def schema_constraint(cfg):
 
     return cfg
 
+############################################
+# Server Configuration
+############################################
+
+def schema_server(cfg):
+
+    scparam(cfg, ['server', 'jobscheduler'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Job scheduler name",
+            switch="-server_jobscheduler <str>",
+            example=[
+                "cli: -server_jobscheduler slurm",
+                "api: chip.set('server','jobscheduler','slurm')"],
+            schelp="""
+            Sets the type of job scheduler to be used for each individual
+            flowgraph steps. If the parameter is undefined, the steps are executed
+            on the same machine that the SC was launched on. If 'slurm' is used,
+            the host running the 'sc' command must be running a 'slurmctld' daemon
+            managing a Slurm cluster. Additionally, the build directory ('-dir')
+            must be located in shared storage which can be accessed by all hosts
+            in the cluster.""")
+
+    scparam(cfg, ['server', 'cores'],
+            sctype='int',
+            scope='job',
+            shorthelp="Server: Core constraint",
+            switch="-server_cores <int>",
+            example= ["cli: -server_cores 48",
+                    "api: chip.set('server', 'cores', '48')"],
+            schelp="""
+            Specifies the number cpu cores required to run the job.
+            For the slurm scheduler, this translates to the '-c'
+            switch.""")
+
+    scparam(cfg, ['server', 'memory'],
+            sctype='int',
+            unit='MB',
+            scope='job',
+            shorthelp="Server: Memory constraint",
+            switch="-server_memory <str>",
+            example= ["cli: -server_memory 8000",
+                    "api: chip.set('server', 'memory', '8000')"],
+            schelp="""
+            Specifies the amount of memory required to run the job,
+            specified in MB. For the slurm scheduler, this translates to
+            the '--mem' switch. """)
+
+    scparam(cfg, ['server', 'constraint'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Machine constraint",
+            switch="-server_constraint <str>",
+            example= [
+                "cli: -server_constraint 'riscv&gpu'",
+                "api: chip.set('server', 'constraint', 'riscv&gpu')"],
+            schelp="""
+            Provides machine feature constraints to the job scheduler for
+            selecting compute nodes. Feature names are strings assigned to
+            computes nodes by the job scheduler administrator. The constraint
+            parameter is passed through to the job schedular unmodified. For
+            the slurm scheduler, this translates to the '--constraint'
+            switch. See the slurm documentation for examples of the types of
+            constraints supported.""")
+
+    scparam(cfg, ['server', 'queue'],
```

Comment:
Should we call it `partition`? Could be confusing since Slurm has a command called `squeue`.

Reply:
Queue is the term that people are familiar with from chip design and to me a clearer name. If we are going to support all three, then we will have translation/confusion regardless. I consider queue a better word than partition.
https://slurm.schedmd.com/rosetta.html

Reply:
I would prefer `queue`